### DISCUSSION TITLE
Feature request: add Loudness Equalization toggle

### DISCUSSION THREAD

**Discussion by SamiPirbay** [File-New-Project/EarTrumpet#656] [Ideas]
Loudness Equalization can be found in Speakers Properties &rightarrow; Enhancements. It's a bit of a trek to get there.

It's useful when watching Netflix if you don't want action sequences to be louder than the rest of the movie, for instance.

**Comment by riverar** [comment]
Sounds like a cool idea; will investigate exposing enhancements

**Reply by ghost** [reply]
You're the best!

**Comment by Riajyuu** [comment]
Just a quick note: some laptop doesn't have this in system settings.

**Comment by fsfod** [comment]
I managed to get this working in my branch here https://github.com/fsfod/EarTrumpet/commits/loudness_fx i added it as a checkbox in the right click menu that changes it for the default device. Tthe tldr of it is setting a FX config option with IPolicyConfigWin7.SetPropertyValue with the guid of E0A941A0-88A2-4df5-8D6B-DD20BB06E8FB and property id of 4

**Comment by SamiPirbay** [comment]
It doesn't work on my computers. I believe this GUID is specific to Realtek High Definition Audio devices.

It's still useful since Realtek is in so many systems, but I think the UI should somehow warn the user or disable the checkbox in unsupported systems

**Comment by fsfod** [comment]
You could maybe find the FX property store GUID for your device using some registry monitor and toggling the option in the windows property page just filter to a registry path containing "FxProperties". My code could just be toggling it on the wrong output device you idk.

**Comment by SamiPirbay** [comment]
I did a three-way registry diff (original state, state after toggle, and state after toggle back) and found that I can do it by changing the key `{fc52a749-4be9-4510-896e-966ba6525980},3` from `xx xx xx xx xx xx xx xx - 00 00 xx xx` to `xx xx xx xx xx xx xx xx - FF FF xx xx`. More obscure than a 0 to 1 switch but it works. I googled that key and found that it's called "MFPKEY_CORR_LOUDNESS_EQUALIZATION_ON".

So I can adapt your code for my computer, which is awesome, thanks :)

But how to make theses changes safe and generic enough for the release version?

**Reply by ghost** [reply]
> 
> 
> I did a three-way registry diff (original state, state after toggle, and state after toggle back) and found that I can do it by changing the key `{fc52a749-4be9-4510-896e-966ba6525980},3` from `xx xx xx xx xx xx xx xx - 00 00 xx xx` to `xx xx xx xx xx xx xx xx - FF FF xx xx`. More obscure than a 0 to 1 switch but it works. I googled that key and found that it's called "MFPKEY_CORR_LOUDNESS_EQUALIZATION_ON".
> 
> So I can adapt your code for my computer, which is awesome, thanks :)
> 
> But how to make theses changes safe and generic enough for the release version?

Surely EarTrumpet already knows the GUID of the current device as it knows how to switch them?
Shouldn't this be fairly easy to implement in the main branch here?

**Comment by ghost** [comment]
_sorry, posted in a new discussion having missed this one_
from previous post:

So many audio sources have their volumes all messed up, and when the source is not super important (to be kept as is), loudness equalization helps a lot.
However, switching it on/off is a pain:

Right-click EarTrumpet > Windows Legacy > Playback Devices > Right-click Device > Properties > Enhancements > Loudness Equalization > OK > OK 

![image](https://user-images.githubusercontent.com/17074394/136942646-b849e70f-a83c-4d4b-85d8-f9e8db9fa0ae.png)



I don't know how much work it is to get access to this function but I did some research and it seems to work though a registry key, and then resetting the audio driver for the change to have effect.

Example:
```
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\MMDevices\Audio\Render\{someIDforTheDevice}\FxProperties, {E0A941A0-88A2-4df5-8D6B-DD20BB06E8FB},4`
```

So that key has to change and audio driver or something has to reset/restart for it to apply the change.
